src/kernels.jl

```diff
@@ -194,6 +205,13 @@ function _transduce!(buf, rf::F, init, arrays...) where {F}
     return dest, buf
 end
 
+# Since CUDA already requires that everything is inlined, `restack` is not
+# useful.  Instead, it's better to avoid introducing extra function calls to
+# reduce the change that inliner gives up.
+@static if isdefined(Transducers, :restack) && isdefined(CUDA, Symbol("@device_override"))
+    CUDA.@device_override Transducers.restack(x) = x
+end
```

Comment:
Since https://github.com/JuliaFolds/Transducers.jl/pull/504 didn't work, this is the next easiest way to do it